docs/new-feature-tutorial.md

```diff
@@ -150,102 +172,183 @@ send some response to the client that made the request.
-
-In this case you don't need to change `apply_event` because there is
-already code that will correctly handle the realm update event type: :
+      if want('realm'):
+        for property_name in Realm.property_types:
+            state['realm_' + property_name] = getattr(user_profile.realm, property_name)
+        state['realm_authentication_methods'] = user_profile.realm.authentication_methods_dict()
+        state['realm_allow_message_editing'] = user_profile.realm.allow_message_editing
+        # ...
 
     def apply_event(state, events, user_profile, include_subscribers):
       for event in events:
         # ...
         elif event['type'] == 'realm':
            field = 'realm_' + event['property']
            state[field] = event['value']
+           # ...
+
+If you are adding a realm property that fits the `do_set_realm_property`
+framework, you don't need to change `fetch_initial_state_data` or
+`apply_event` because there is already code to get the initial data and
+the realm update event type. However, if you are adding a property
+that is handled separately, you will need to  explicitly add the property
+to the `state` dictionary in the `fetch_initial_state_data` function.
+Ex, for `authentication_methods`:
+
+    def fetch_initial_state_data(user_profile, event_types, queue_id, include_subscribers=True):
+      # ...
+      if want('realm'):
+          # ...
+          state['realm_authentication_methods'] = user_profile.realm.authentication_methods_dict()
+          # ...
+
+You will not need to change `apply_event`.
 
 ### Add a new view
 
-You then need to add a view for clients to access that will call the
-newly-added `actions.py` code to update the database. This example
-feature adds a new parameter that should be sent to clients when the
-application loads and be accessible via JavaScript, and there is already
-a view that does this for related flags: `update_realm`. So in this
-case, we can add our code to the existing view instead of creating a
-new one. :
+You will need to add a view for clients to access that will call the
+`actions.py` code to update the database. This example feature
+adds a new parameter that should be sent to clients when the
+application loads and should be accessible via JavaScript. There is
+already a view that does this for related flags: `update_realm` in
+`zerver/views/realm.py`. So in this case, we can add our code to the
+existing view instead of creating a new one.
 
-    # zerver/views/home.py
+First, add the new feature to the `page_params_core_fields` list
+in `zerver/views/home.py`.
 
-    def home(request):
+    def home_real(request):
       # ...
-      page_params = dict(
+      page_params_core_fields = [
         # ...
-        realm_invite_by_admins_only = register_ret['realm_invite_by_admins_only'],
+        'realm_icon_url',
+        'realm_invite_by_admins_only',
+        'realm_inline_image_preview',
```

Comment:
actually that might not be really worth it. didn't realize how separated this was from the `page_param_core_fields`